### PR TITLE
feat(auth): added api endpoint for AuthModule

### PR DESCRIPTION
**Pull request by yshashi**
## Release Notes

- **New Features**
  - Introduced JWT authentication and GitHub OAuth integration.
  - Added endpoints for user registration, login, token refresh, and GitHub login.
  - New data structures for managing authentication requests and responses.

- **Enhancements**
  - Updated user model to include authentication-related fields (e.g., PasswordHash, IsEmailVerified, RefreshToken).
  - Improved configuration settings for JWT and GitHub authentication.
  - Enhanced API documentation with authentication details.

- **Bug Fixes**
  - Resolved issues related to authentication flow and token management.

These changes enhance user experience by providing secure authentication methods and streamlined access to features.

### CODE REVIEW

File: devs-who-run-api/Modules/Auth/Services/AuthService.cs [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+public class GitHubUser
+{
+    public string? Login { get; set; }
+    public string? Id { get; set; }
+    public string? Name { get; set; }
+    public string? Email { get; set; }
+    public string? AvatarUrl { get; set; }
+}
```
Comment (santoshyadavdev) [inline devs-who-run-api/Modules/Auth/Services/AuthService.cs:231]: should we keep this in a separate file?

File: devs-who-run-api/appsettings.json [hunk 2]
```diff
@@ -5,16 +5,28 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
+  "AllowedHosts": "*",
   "Database": {
-    "ConnectionString": "Server=localhost;Port=5432;Database=devswhorun;User Id=postgres;Password=admin;",
+    "ConnectionString": "Host=localhost;Port=5432;Database=devswhorun;Username=postgres;Password=postgres",
```
Comment (santoshyadavdev) [inline devs-who-run-api/appsettings.json:10]: any reason this is changed?

**Review by santoshyadavdev — APPROVED** [review]
small comment

File: devs-who-run-api/appsettings.json (hunk 2 above)
Comment (yshashi) [reply devs-who-run-api/appsettings.json:10]: Nice catch, yes I need to remove this one as this is my machine's configuration

File: devs-who-run-api/Modules/Auth/Services/AuthService.cs (hunk 1 above)
Comment (yshashi) [reply devs-who-run-api/Modules/Auth/Services/AuthService.cs:231]: I kept this intentionally for easy reference, will move them into separate file after github auth integration

**Pull request merged**